### PR TITLE
Gas Masks filtering and tint update

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This is the atomized version of the gas mask filtering from #56999 
This PR allows gas masks to properly filter toxic gases by using one-time-use filters that can be made in any lathe from roundstart. To counter this now gas masks darken the field of view when worn. I also updated the tint calculation to allow more numbers other than 1, 2 (all numbers between 0 and 3 are now fine and won't totally blind you, but the effect stacks)

## Why It's Good For The Game
Gas masks working properly, wearing a mask that filter gases and hide your face should come with a downside

## Changelog
:cl:
balance: gasmasks now properly filter gases by using craftable filters but darken your fov
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
Can you know how long it'll last for? As though you were wearing a watch on the inside of your wrist?

**Comment by Ghilker** [comment]
if you mean the duration in minutes, i don't know, because it depends on the amount of gases and on how many different gases are in the atmosphere, so it would be difficult to code too, but you can examine the mask and it will tell you the remaining % of the filters

**Comment by cacogen** [comment]
I really just meant knowing when it would run out. That's good.

**Comment by Sparkezel** [comment]
Bro, what about sec masks?

**Comment by Ghilker** [comment]
> Bro, what about sec masks?

what about them?

**Comment by Sparkezel** [comment]
well, they still mask your identity and face despite covering only your mouth sprite-wise iirc

**Comment by Ghilker** [comment]
I don't know all masks/variations, so if there is a mask that doesn't cover eyes/doesn't give unkown status, lemme know

**Comment by Sparkezel** [comment]
nothing other comes to my mind so I think secmask needs a look, thats all

**Comment by maxymax13** [comment]
can you still use internals with them? if you can, they better not waste filters while using them as internals

**Comment by Ghilker** [comment]
> can you still use internals with them? if you can, they better not waste filters while using them as internals

you can, in theory with internals the filter doesn't get consumed, but i havent tested that

**Comment by Rustybuckets6601** [comment]
Finally, Metro gameplay

**Comment by CRITAWAKETS** [comment]
Are you my mummy?

**Comment by ATH1909** [comment]
> I really just meant knowing when it would run out. That's good. I always wondered why these didn't block gases like you'd expect.

for what it's worth, they block the inhalation of chem smoke without needing you to turn on internals

**Comment by ATH1909** [comment]
would it be possible to make it so that you can use wirecutters or something on a gas mask to turn it into a version of the mask without filters or tint, for people who just want to wear a gas mask for cosmetic reasons/:soul:?

**Comment by RaveRadbury** [comment]
Ghilker, consider atomizing the filters and the tint.

**Comment by Coastfront** [comment]
Metro 13

**Comment by Tlaltecuhtli** [comment]
can these have actual utilities instead of filtering gases that internals are 1000% better at it?

**Comment by Ghilker** [comment]
> Ghilker, consider atomizing the filters and the tint.

ehh i think the two should go in the same pr since is a rebalance on how gasmasks works

**Comment by YakumoChen** [comment]
I feel pretty bad about this. A lot of the reason to wear a gas mask is to hide identity, and this steps on a lot more toes than just making bad gas less bad for awhile.

I don't personally feel the positive changes outweigh the many seen and unseen negatives put into play by nerfing gas masks so much.

How does this interact with Chameleon masks by the way?

**Comment by Ghilker** [comment]
> I feel pretty bad about this. A lot of the reason to wear a gas mask is to hide identity, and this steps on a lot more toes than just making bad gas less bad for awhile.
> 
> I don't personally feel the positive changes outweigh the many seen and unseen negatives put into play by nerfing gas masks so much.
> 
> How does this interact with Chameleon masks by the way?

Being able to hide your identity without any drawback is bad

Will have to test on chameleon masks

**Comment by ATH1909** [comment]
> Being able to hide your identity without any drawback is bad

bruh

there are SO many ways in this game to conceal your face. you can literally get a face-concealing mask from the mats used to make a single wooden table. there are plenty of helmets and stuff that can coneal your face without even taking up your mask slot.

**Comment by Ghilker** [comment]
Wearing a gasmask only

Wearing a gasmask and a pair of sunglasses

**Comment by ATH1909** [comment]
how does this interact with welding gas masks?

**Comment by Ghilker** [comment]
> how does this interact with welding gas masks (which are really more like welding goggles that go in your glasses slot instead of your eyewear slot than they are like gas masks)?

is similar to the darkest one when you have the welding glass on, if you remove it it goes away

**Comment by Livrah** [comment]
Gasmask+sunglasses HUD doesnt make any sense.

**Comment by Ghilker** [comment]
> Gasmask+sunglasses HUD doesnt make any sense.

sunglasses makes your vision darker, if you add a gasmask on top of it...

**Comment by Fikou** [comment]
this is just shitting on random masks, if i use some masks i get a huge disadvantage but if i use other ones i can wear sunglasses just fine, you started using decimals for the tint for some reason, and the mask or glasses themselves doesnt matter other than slightly inconvieniencing players with darkness

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -644,12 +644,23 @@
...
 	else if(tinttotal >= TINT_DARKENED)
 		cure_blind(EYES_COVERED)
-		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 2), clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal >= TINT_LIGHTER)
+		cure_blind(EYES_COVERED)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 1), clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal > TINT_LIGHT)
+		cure_blind(EYES_COVERED)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 1), clamp(tinttotal * 100, 0, 255))
```
Comment (Fikou) [inline code/modules/mob/living/carbon/carbon.dm:663]: arent these the same exact thing

Comment (Ghilker) [reply code/modules/mob/living/carbon/carbon.dm:663]: no because the tinttotal var get passed as alpha, so in the first one, you get darker alpha, in the second one, you get it ligher

Comment (Fikou) [reply code/modules/mob/living/carbon/carbon.dm:663]: but it passes it anyway?

Comment (Ghilker) [reply code/modules/mob/living/carbon/carbon.dm:663]: yes but the first one is from 1 to 1.99, the second one is from 0.1 to 0.99
currently the second one should be unused tho since there aren't any clothing with less than 1 tint but i added it for modularity

Comment (nightred) [reply code/modules/mob/living/carbon/carbon.dm:663]: But the actual line `overlay_fullscreen`  does not change based on the the tinttotal value, on the value in tinttotal  changes.

Comment (Ghilker) [reply code/modules/mob/living/carbon/carbon.dm:663]: ahh yes i see what you mean, in the second one should be `change_overlay ? 3 : 1` instead of `change_overlay ? 4 : 1` i'll fix it

**Comment by YakumoChen** [comment]
> 
> 
> > I feel pretty bad about this. A lot of the reason to wear a gas mask is to hide identity, and this steps on a lot more toes than just making bad gas less bad for awhile.
> > I don't personally feel the positive changes outweigh the many seen and unseen negatives put into play by nerfing gas masks so much.
> > How does this interact with Chameleon masks by the way?
> 
> Being able to hide your identity without any drawback is bad


The drawback is that it makes you look suspicious.

The drawback of other face hiding masks is that they still easily identify you.

**Comment by Ghilker** [comment]
> The drawback is that it makes you look suspicious.
> 
> The drawback of other face hiding masks is that they still easily identify you.

you can't be suspicious if the majority is wearing one

**Comment by itseasytosee** [comment]
Do you genuinely think gas masks are overpowered because they conceal your identity?

**Comment by Sethnov** [comment]
this may sound counterintuitive but can you remove the glass from a welder helmet, so I can just wear it cosmetically?

**Comment by Fox-McCloud** [comment]
Gas masks hindering your vision is going to lead to this nasty feedback loop, in my opinion.

No one will want to wear them anymore, because of it---and the masks that don't hinder your vision will inherently draw more suspicion; not to mention this makes it even harder to hide your identity.

Perhaps people just wearing gas masks all the time and *constantly* hiding their identity may be somewhat of a problem, but, this makes stealth even more difficult and encourages you to go ham even more, and it also pushes stealth more into the category of "it's possible, but it's clear if you're doing anything stealth/identity hiding related, then you're probably an antag".

Having doubt is probably better than having fore-sure tells.

**Comment by itseasytosee** [comment]
Why do you guys think hiding your face is so powerful? Countless items do it. Gasmasks are just common.

**Comment by cacogen** [comment]
Gas masks by themselves should be darkened around the green circles like in the sunglasses picture, but to a lesser extent. 

If it were me, I'd have the green circles large enough to cover most of your field of view and be cut off slightly at the edges.

**Comment by RaveRadbury** [comment]
It's kind of silly to do this lens-shaped filter, given that we have established that view obfuscation is meant to be done in a circle around the player. If anything, a parabolic shape that has the arc behind the player (to simulate loss of peripheral vision) would be a better representation of what's being attempted here.

**Comment by cacogen** [comment]
I tested this locally. The overlay doesn't update properly (have to take off mask and put it back on).

>It's kind of silly to do this lens-shaped filter, given that we have established that view obfuscation is meant to be done in a circle around the player.

Yeah, that makes more sense.

**Comment by YakumoChen** [comment]
> 
> 
> Why do you guys think hiding your face is so powerful? Countless items do it. Gasmasks are just common.

precisely because they're common, and a lot of people wear them. If you wear a tiki mask or cardborg helmet you're immediately identifiable by it. But "was wearing a gas mask" could describe 20 people currently in the game.

**Comment by Mothblocks** [comment]
sounds to me like theyll just move onto something else

**Comment by YakumoChen** [comment]
I mean, gas masks _are_ common. They're in every fire closet, half the departments on station spawn with one, they're classic greytide wear. I'm not sure why people would switch except possibly because they've been needlessly nerfed.

**Comment by Mothblocks** [comment]
i didn't mean that in a pro-this-pr way to clarify. i meant it in a this feels pointless way

**Comment by RaveRadbury** [comment]
> i didn't mean that in a pro-this-pr way to clarify. i meant it in a this-pr-feels-pointless way

tbh having masks that would obstruct vision do some light vision obstruction is pretty nice, I'd be in favor of the parabolic cone of vision thing

**Comment by moocowswag** [comment]
> precisely because they're common, and a lot of people wear them. If you wear a tiki mask or cardborg helmet you're immediately identifiable by it. But "was wearing a gas mask" could describe 20 people currently in the game.

There are plenty of ways to cover your face to the point where you could constantly switch between them, also there is such a thing as doing your crime while wearing a tiki mask and then ditching it immediately after.

And then there is the fact that disguises can be alot more than just covering your face and people can single you out based on your jumpsuit/exosuit.

Tldr: people can just move on to something else as stealth options and strats are extremely extensive in variety.

**Comment by optimumtact** [comment]
Change it to a circle please, the effect looks weird.

**Comment by Ghilker** [comment]
> Change it to a circle please, the effect looks weird.

Will do

**Comment by LemonInTheDark** [comment]
?
